onnxruntime/core/framework/kernel_type_str_resolver_utils.cc

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+
+#include "core/framework/kernel_type_str_resolver_utils.h"
+
+#include "flatbuffers/flatbuffers.h"
+
+#include "core/common/common.h"
+#include "core/flatbuffers/schema/ort.fbs.h"
+#include "core/optimizer/transpose_optimizer/layout_transformation_potentially_added_ops.h"
+
+namespace onnxruntime::kernel_type_str_resolver_utils {
+
+static constexpr auto* kStandaloneKernelTypeStrResolverFileIdentifier = "ktsr";
+
+#if !defined(ORT_MINIMAL_BUILD)
+
+gsl::span<const OpIdentifierWithStringViews> GetLayoutTransformationRequiredOpIdentifiers() {
+  return kLayoutTransformationPotentiallyAddedOps;
+}
+
+Status SaveKernelTypeStrResolverToBuffer(const KernelTypeStrResolver& kernel_type_str_resolver,
+                                         flatbuffers::DetachedBuffer& buffer, gsl::span<const uint8_t>& buffer_span) {
+  flatbuffers::FlatBufferBuilder builder;
+  flatbuffers::Offset<fbs::KernelTypeStrResolver> fbs_kernel_type_str_resolver;
+  ORT_RETURN_IF_ERROR(kernel_type_str_resolver.SaveToOrtFormat(builder, fbs_kernel_type_str_resolver));
+  builder.Finish(fbs_kernel_type_str_resolver, kStandaloneKernelTypeStrResolverFileIdentifier);
+  buffer = builder.Release();
+  buffer_span = gsl::make_span(buffer.data(), buffer.size());
+  return Status::OK();
+}
+
+#endif  // !defined(ORT_MINIMAL_BUILD)
+
+Status LoadKernelTypeStrResolverFromBuffer(KernelTypeStrResolver& kernel_type_str_resolver,
+                                           gsl::span<const uint8_t> buffer_span) {
+  flatbuffers::Verifier verifier{buffer_span.data(), buffer_span.size_bytes()};
+  ORT_RETURN_IF_NOT(verifier.VerifyBuffer<fbs::KernelTypeStrResolver>(kStandaloneKernelTypeStrResolverFileIdentifier),
+                    "Failed to verify KernelTypeStrResolver flatbuffers data.");
+  const auto* fbs_kernel_type_str_resolver = flatbuffers::GetRoot<fbs::KernelTypeStrResolver>(buffer_span.data());
+  ORT_RETURN_IF_ERROR(kernel_type_str_resolver.LoadFromOrtFormat(*fbs_kernel_type_str_resolver));
+  return Status::OK();
+}
+
+Status AddLayoutTransformationRequiredOpsToKernelTypeStrResolver(KernelTypeStrResolver& kernel_type_str_resolver) {
+  KernelTypeStrResolver resolver_with_required_ops{};
+
+  // to generate kRequiredOpsKernelTypeStrResolverBytes, run the test:
+  //   KernelTypeStrResolverUtilsTest.DISABLED_PrintExpectedLayoutTransformationRequiredOpsResolverByteArray
+
+  // clang-format off
+  constexpr uint8_t kLayoutTransformationRequiredOpsKernelTypeStrResolverBytes[] = {
```

Comment:
Are these replacements for hash values ? :)

Reply:
right, it effectively is a replacement for these hardcoded hashes

https://github.com/microsoft/onnxruntime/blob/64466c2d6206d25de52222a49f0cda0f0a145787/onnxruntime/core/framework/kernel_def_hash_helpers.cc#L11-L66